dependency issue: sqlparse 0.4.1-1

## Description
After updating **sqlparse** to 0.4.1-1, **pgcli** fails to run because of dependency issue.
It's obvious, because in **pgcli**'s requirement **sqlparse** is written as `sqlparse<4.0,>=0.3.0`.

I've tried to drop this restriction but then a couple of tests are failed:
`tests/test_smart_completion_public_schema_only.py`:
```python
test_specials_included_for_initial_completion

initial_text = '    '

    @pytest.mark.parametrize("initial_text", ("", "    ", "\t \t"))
    def test_specials_included_for_initial_completion(initial_text):
>       suggestions = suggest_type(initial_text, initial_text)

tests/test_sqlcompletion.py:821:

../../pgcli/packages/sqlcompletion.py:142: in suggest_type
    stmt = SqlStatement(full_text, text_before_cursor)
../../pgcli/packages/sqlcompletion.py:59: in __init__
    full_text, text_before_cursor, self.local_tables = isolate_query_ctes(
../../pgcli/packages/parseutils/ctes.py:23: in isolate_query_ctes
    ctes, remainder = extract_ctes(full_text)
_ _ _ 

sql = '    '

    def extract_ctes(sql):
        """ Extract constant table expresseions from a query

            Returns tuple (ctes, remainder_sql)

            ctes is a list of TableExpression namedtuples
            remainder_sql is the text from the original query after the CTEs have
            been stripped.
        """

>       p = parse(sql)[0]
E       IndexError: tuple index out of range

../../pgcli/packages/parseutils/ctes.py:58: IndexError
```

Of course the problem is gone if I use a separate venv, but I prefer to use AUR's version of **pgcli** to make it system-wide.
And because of rolling-release it's unconvinient to keep **sqlparse** outdated.
I hope this trouble with deps could be fixed)

## Your environment
ArchLinux 5.8.7-arch1-1 x86_64
bash 5.0.018-1
pgcli 3.0.0-1 (from AUR)
python-sqlparse 0.4.1-1 (from cimmunity)

--- a/AUTHORS
+++ b/AUTHORS
@@ -113,6 +113,7 @@ Contributors:
     * Seungyong Kwak (GUIEEN)
     * Tom Caruso (tomplex)
     * Jan Brun Rasmussen (janbrunrasmussen)
+    * Kevin Marsh (kevinmarsh)
 
 Creator:
 --------

--- a/changelog.rst
+++ b/changelog.rst
@@ -9,6 +9,7 @@ Features:
 * Update python version in Dockerfile
 * Support setting color for null, string, number, keyword value
 * Support Prompt Toolkit 2
+* Support sqlparse 0.4.x
 * Update functions, datatypes literals for auto-suggestion field
 * Add suggestion for schema in function auto-complete
 

--- a/pgcli/packages/parseutils/ctes.py
+++ b/pgcli/packages/parseutils/ctes.py
@@ -16,7 +16,7 @@ TableExpression = namedtuple("TableExpression", "name columns start stop")
 def isolate_query_ctes(full_text, text_before_cursor):
     """Simplify a query by converting CTEs into table metadata objects"""
 
-    if not full_text:
+    if not full_text or not full_text.strip():
         return full_text, text_before_cursor, tuple()
 
     ctes, remainder = extract_ctes(full_text)

--- a/setup.py
+++ b/setup.py
@@ -13,7 +13,7 @@ install_requirements = [
     # see: https://github.com/dbcli/pgcli/pull/1197
     "prompt_toolkit>=2.0.6,<4.0.0",
     "psycopg2 >= 2.8",
-    "sqlparse >=0.3.0,<0.4",
+    "sqlparse >=0.3.0,<0.5",
     "configobj >= 5.0.6",
     "pendulum>=2.1.0",
     "cli_helpers[styles] >= 2.0.0",

--- a/tests/test_sqlcompletion.py
+++ b/tests/test_sqlcompletion.py
@@ -816,7 +816,7 @@ def test_create_db_with_template():
     assert set(suggestions) == set((Database(),))
 
 
-@pytest.mark.parametrize("initial_text", ("", "    ", "\t \t"))
+@pytest.mark.parametrize("initial_text", ("", "    ", "\t \t", "\n"))
 def test_specials_included_for_initial_completion(initial_text):
     suggestions = suggest_type(initial_text, initial_text)
 

